script.js

```diff
@@ -0,0 +1,117 @@
+
+function Location() {
+  navigator.geolocation.getCurrentPosition(gotLocation);
+}
+
+Location();
+
+function gotLocation(position) {
+
+  const latitude = position.coords.latitude
+  const longitude = position.coords.longitude
+
+  const apiKey = "82005d27a116c2880c8f0fcb866998a0";
+  const apiUrl = `https://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=${apiKey}`;
+
+  async function currentLocationData() {
+    try {
+      document.getElementById('image').src = "loader.gif";
+      const res = await fetch(apiUrl);
+      const data = await res.json();
+      const tempInCalvin = data.main?.temp;
+      let tempInDegree = tempInCalvin - 273.15;
+      tempInDegree = Math.round(tempInDegree);
+      const iconCode = data.weather[0]?.icon;
+      const iconUrl = `https://openweathermap.org/img/w/${iconCode}.png`;
+      document.getElementById('image').src = iconUrl;
+      document.getElementById("desc").innerHTML = data.weather[0]?.description
+      document.getElementById("address").innerHTML = data?.name + ", ";
+      document.getElementById("country").innerHTML = data?.sys.country;
+      document.getElementById("temperature").innerHTML = tempInDegree + "°c";
+
+
+    } catch (err) {
+      console.log(err)
+    }
+
+  }
+  currentLocationData();
+}
+
+
+
+
+const helperFunction = (city) => {
+  return `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=82005d27a116c2880c8f0fcb866998a0`
+}
+
+// const handleSearch = async () => {
+async function handleSearch() {
+  var data;
+  // document.getElementById("error").style.display = "none"
+
+
+  try {
+    document.getElementById('image').src = "loader.gif";
+    document.getElementById("weatherContainer").style.display = "flex"
+    document.getElementById("error").style.display = "none"
+    let inputValue = document.getElementById("input").value;
+    const response = await fetch(helperFunction(inputValue));
+    data = await response.json();
+
+    const tempInCalvin = data.main?.temp;
+    let tempInDegree = tempInCalvin - 273.15;
+    tempInDegree = Math.round(tempInDegree);
+    const iconCode = data.weather[0]?.icon;
+    const iconUrl = `https://openweathermap.org/img/w/${iconCode}.png`;
```

Comment:
repeated code, create a function for it and do these from there.
const tempInCalvin = data.main?.temp;
    let tempInDegree = tempInCalvin - 273.15;
    tempInDegree = Math.round(tempInDegree);
    const iconCode = data.weather[0]?.icon;
    const iconUrl = `https://openweathermap.org/img/w/${iconCode}.png`;
document.getElementById('image').src = iconUrl;
    document.getElementById("desc").innerHTML = data.weather[0]?.description
    document.getElementById("address").innerHTML = data?.name + ", ";
    document.getElementById("country").innerHTML = data?.sys.country;
    document.getElementById("temperature").innerHTML = tempInDegree + "°c";